data-preprocessing/webtextbook_parsing_script/extract_textbook_from_url_notebook_.ipynb

```diff
@@ -0,0 +1,462 @@
+    "\n",
+    "3. extract_url(link, pageList, maxNmber, char): It extracts from the list of the table of content and concatenate it with the link to form the desired page URL\n",
+    "\n",
+    "*         link: just as described in the function above.\n",
+    "\n",
+    "*         pageList: A list containing the table of contents extracted from the previous function\n",
+    "        \n",
+    "*         maxNmber:  The maximum index number on the table of content of the desired textbook\n",
+    "\n",
+    "*         char(Optional): A list containing the first character of the index in the table of content that is not a number\n",
+    "\n",
+    "4. extract_url_content(url, file_name): Takes the textbook's URL and name of the textbook, extracts and structures the data from the URL, and stores it with the file name. NB: file_name is optional.\n",
+    "\n",
+    "5. extract_textbook(url_list, textbook_name): Iterates over the list of URL list, generated by function 3, parse it into function 4 to extract the contents, and then save it as a JSON file with the name of the textbook 'textbook_name'.\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Extract URL for each pages of the Textbook\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "#-----------Function: Remove dublicates-------------------------#\n",
+    "\n",
+    "def unique(array):\n",
+    "# This funtion takes a list and remove dublicates\n",
+    "    return list(dict.fromkeys(array))\n",
+    "\n",
+    "\n",
+    "def remove_items(a, b):\n",
+    "    # this function removes a subset from a supperset of a list.\n",
+    "    # it was used to filter out Paaragraph_Not_in_Section\n",
+    "    # a = supperset = referring to all paragraphs (Para) , b = subset = referring to all sectioned paragraphs\n",
+    "    a = [item for item in a if item not in b]\n",
+    "    return a\n",
+    "\n",
+    "#---------------------------------------------------------------#\n",
+    "\n",
+    "#-----------Function: Extract table of contents -----------------#\n",
+    "import requests\n",
+    "from bs4 import BeautifulSoup\n",
+    "\n",
+    "import os\n",
+    "from urllib.parse import urljoin\n",
+    "import json\n",
+    "\n",
+    "def extract_rawTable_of_content(link, homePage):\n",
+    "\n",
+    "    # --> homePage = First landing pages of online view of the textbook(eg: \"1-introduction\")\n",
+    "    # ---> link = url of the site excluding landing page indexing (eg: \"https://openstax.org/books/university-physics-volume-3/pages/\")\n",
+    "\n",
+    "    website_link = link+homePage\n",
```

Comment:
if this is the only time `link` and `homePage` are used here may have been worth just passing in a single argument `website_link` to the function instead of the 2 arguments. Then at the function usage do the appending